### DISCUSSION TITLE
KeyValue history limited to 64 - why?

### DISCUSSION THREAD

**Discussion by simonhoss** [nats-io/nats.net#351] [General]
Hey

I wonder why the KeyValue history is limited to 64? Is there any bigger reason behind it? Is it possible to override? We would need much more.

The limitation is set here: https://github.com/nats-io/nats.net.v2/blob/6b22255373a810c1a3fa9b260a1266f6dffc4a99/src/NATS.Client.KeyValueStore/NatsKVDefaults.cs#L5

Greets

**Comment by mtmk** [comment]
hey @simonhoss, not sure why. I think it comes from the spec/ADR https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-8.md let me see if I can find out why 😅

**Reply by mtmk** [reply]
sorry @simonhoss it's actually by design as the 'KV API' and is set in all NATS clients. The idea is to not restrict future development of KV features for example where it might not be backed by a stream but potentially a new storage engine implemented in nats-server. You should use a stream if you need unlimited history.

**Reply by simonhoss** [reply]
Got it. Thanks
